local-startup/running-proxy-node-locally.md

```diff
@@ -0,0 +1,81 @@
+
+- The VSP runs in `development` mode which means it talks to the IDA Compliance Tool which acts as a pretend Verify Hub. The Compliance Tool doesn't generate a SAML message but a number of URLs to simulate Hub responses. None of those responses contain any assertions, though, so the translation functionality is not tested in a local journey. The next step would be to either spin up the Hub locally and connect to that but that's computationally expensive. An alternative may be to try to run the Compliance Tool locally and configure it to return SAML messages with assertions.
+
+- The PKI and metadata files are provided statically rather being generated dynamically. This means the provided signed metadata files and all the keys and certs can't be changed easily. This is okay for now but in the future we'd want to review the existing PKI and metadata generation scripts and use them for local running.
+
+    The metadata xml files have been manually signed with `xmlsectool` and cannot be changed without re-signing. The metadata truststores have been generated manually with `keytool` and need to be re-generated if metadata gets re-signed with a different cert chain. The certs and private keys used by the apps are symlinked from `verify-dev-pki`. 
+
+## Building images
+
+There are two ways to build Docker images - compile code in Docker (slower) or use locally built artifacts. By default, Docker will compile code as part of the build step. This has the disadvantage that gradle cache is unavailable in the Docker image so all dependencies have to be downloaded each time for every app and everything has to be compiled from scratch. This makes the build phase quite slow.
+
+The alternative is to set the `USE_LOCAL_BUILD` environment variable to instruct the Docker image to copy locally built artifacts from `${component}/build/install`. To produce the artifacts run `./gradlew installDist [-p ${component}]` to build distribution binaries [for a specific app] before building with Docker. This takes full advantage of gradle's caching mechanisms and makes re-building images much faster.
+
+You can also set the `VERIFY_USE_PUBLIC_BINARIES` variable to use bintray if you're not on the office network and don't want to connect to the VPN.
+
+**Build images and compile code from scratch**
+```shell script
+# Build all apps
+$ docker-compose build [--parallel]
+
+# Build a single app
+$ docker-compose build SERVICE
+
+# For example
+$ docker-compose build translator
+```
+
+**Build images using locally built artifacts**
+```shell script
+# Tell Dockerfile to copy artifacts instead of building them
+$ export USE_LOCAL_BUILD=true
```

Comment:
If `USE_LOCAL_BUILD` only applies to `docker-compose`, you can run it as

`USE_LOCAL_BUILD=true docker-compose ...` which will avoid polluting your shell environment